### PR TITLE
fix: project creation

### PR DESCRIPTION
**Pull request by Jisay**
ref: 
* MANAGER-2745
* MANAGER-2992
* MANAGER-2993

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -3,10 +3,6 @@
   "pci_projects_project_creating_secondary_title": "La création de votre projet est en cours",
   "pci_projects_project_creating_explain1": "Pour finaliser la création de votre projet, veuillez effectuer le paiement du bon de commande <a href=\"{{ href }}\">{{ orderId }}</a>.",
   "pci_projects_project_creating_explain2": "Si cette page est toujours présente plusieurs minutes après le paiement du bon de commande, merci de contacter <a href=\"{{ href }}\">notre service client</a>.",
-  "pci_projects_project_creating_not_paid_title": "En route pour l'aventure Public Cloud",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json]: It seems that this translation key is still used.

Are you sure about removing `pci_projects_project_creating_not_paid_title`?

Please refer to this following file: https://github.com/ovh-ux/manager/pull/985/files#diff-19dfc170661c5c66ec2e6b4028bf87dfR18

File: packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -3,10 +3,6 @@
...
-  "pci_projects_project_creating_not_paid": "Il semble que vous ayez un bon de commande de {{ amount }} de crédits en attente. Nous vous invitons à le régler en cliquant sur \"Finaliser ma commande\" pour lancer la création de votre projet.",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json]: Same for `pci_projects_project_creating_not_paid`.

Please refer to: https://github.com/ovh-ux/manager/blob/develop/packages/manager/modules/pci/src/projects/project/creating/notPaid/notPaid.html#L30.

File: packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -3,10 +3,6 @@
...
-  "pci_projects_project_creating_not_paid_explain1": "En cliquant sur « Finaliser ma commande » vous allez être redirigé sur un bon de commande qu’il faudra régler par le moyen de paiement que vous aurez choisi.",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json]: Same for `pci_projects_project_creating_not_paid_explain1`.

Please refer to: https://github.com/ovh-ux/manager/blob/develop/packages/manager/modules/pci/src/projects/project/creating/notPaid/notPaid.html#L36.

File: packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -3,10 +3,6 @@
...
-  "pci_projects_project_creating_not_paid_explain2": "Pour retrouver votre facture, rendez-vous dans « Mon compte », puis <a href=\"{{ href }}\">« Mes commandes »</a>.",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json]: Same for `pci_projects_project_creating_not_paid_explain2`.

Please refer to: https://github.com/ovh-ux/manager/blob/develop/packages/manager/modules/pci/src/projects/project/creating/notPaid/notPaid.html#L38.

File: packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json (hunk 1 above)
Comment (Jisay) [reply packages/manager/modules/pci/src/projects/project/creating/translations/Messages_fr_FR.json]: Don't worry, there are present ;-) https://github.com/ovh-ux/manager/blob/master/packages/manager/modules/pci/src/projects/project/creating/notPaid/translations/Messages_fr_FR.json#L2

File: packages/manager/modules/pci/src/projects/new/new.routing.js [hunk 5]
```diff
@@ -60,10 +60,16 @@ export default /* @ngInject */ ($stateProvider) => {
...
-        getStateLink: /* @ngInject */ ($state, getCurrentStep) => (action, inherit = true) => {
+        getStateLink: /* @ngInject */ (
+          $state,
+          getCurrentStep,
+          projects,
+        ) => (action, inherit = true) => {
           switch (action) {
             case 'cancel':
-              return $state.href('pci.projects');
+              return $state.href(projects.length
```
Comment (cbourgois) [inline packages/manager/modules/pci/src/projects/new/new.routing.js]: Why not redirecting to `pci.project` and add a `redirectTo` in the state ?
Like done in `pci.projects.project.instances` for example https://github.com/ovh-ux/manager/blob/master/packages/manager/modules/pci/src/projects/project/instances/instances.routing.js#L5

IMO, it's not the role of an external page to detect if we should redirect to the onboarding or the list. We want to go to the list, and if there is no project the list redirects to the onboarding.

**Comment by ovh-cds** [comment]
CDS Report telecom-task#3645.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report freefax#3645.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

File: packages/manager/modules/pci/src/projects/projects.routing.js [hunk 6]
```diff
@@ -10,6 +10,17 @@ export default /* @ngInject */ ($stateProvider) => {
       controller,
       controllerAs: '$ctrl',
       template,
+      redirectTo: (transition) => {
+        const projectsPromise = transition.injector().getAsync('projects');
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/projects.routing.js]:
```suggestion
        =>  transition
            .injector()
            .getAsync('projects')
            .then((projects) => projects.length
                ? true
                : 'pci.projects.onboarding')
```

**Pull request merged**